src/adagrams.js

```diff
@@ -1,15 +1,156 @@
+const LETTER_POOL = {
+	A: 9,
+	B: 2,
+	C: 2,
+	D: 4,
+	E: 12,
+	F: 2,
+	G: 3,
+	H: 2,
+	I: 9,
+	J: 1,
+	K: 1,
+	L: 4,
+	M: 2,
+	N: 6,
+	O: 8,
+	P: 2,
+	Q: 1,
+	R: 6,
+	S: 4,
+	T: 6,
+	U: 4,
+	V: 2,
+	W: 2,
+	X: 1,
+	Y: 2,
+	Z: 1,
+};
+
 export const drawLetters = () => {
-  // Implement this method for wave 1
+	const playersHand = [];
+	const letterBag = [];
+
+	for (const [key, value] of Object.entries(LETTER_POOL)) {
+		for (let i = 0; i < value; i++) {
+			letterBag.push(key);
+		}
+	}
+
+	while (playersHand.length < 10) {
+		let randomNum = Math.floor(Math.random() * letterBag.length);
+		let letterForHand = letterBag[randomNum];
+		playersHand.push(letterForHand);
+		letterBag.splice(randomNum, 1);
+	}
+
+	console.log(playersHand);
+	return playersHand;
 };
 
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+	let word = input.toUpperCase();
+	let result = true;
+	let lettersInHandCopy = [...lettersInHand];
+
+	for (const letter of word) {
+		if (lettersInHand.includes(letter)) {
+			if (lettersInHandCopy.includes(letter)) {
+				let removeLetter = lettersInHandCopy.indexOf(letter);
+				lettersInHandCopy.splice(removeLetter, 1);
+			} else {
+				result = false;
+			}
+		} else {
+			result = false;
+		}
+	}
+	return result;
 };
 
 export const scoreWord = (word) => {
-  // Implement this method for wave 3
+	const scoreChart = {
+		A: 1,
+		B: 3,
+		C: 3,
+		D: 2,
+		E: 1,
+		F: 4,
+		G: 2,
+		H: 4,
+		I: 1,
+		J: 8,
+		K: 5,
+		L: 1,
+		M: 3,
+		N: 1,
+		O: 1,
+		P: 3,
+		Q: 10,
+		R: 1,
+		S: 1,
+		T: 1,
+		U: 1,
+		V: 4,
+		W: 4,
+		X: 8,
+		Y: 4,
+		Z: 10,
+	};
+
+	let points = 0;
+	if (word.length === 0) {
+		return 0;
+	} else {
+		for (let i = 0; i < word.length; i++) {
+			word = word.toUpperCase();
+			if (word.length > 6) {
+				points = 8;
+			} else {
+				points = 0;
+			}
+			for (let letter of word) {
+				let letterValue = scoreChart[letter];
+				points += letterValue;
+			}
+			return points;
+		}
+	}
 };
 
 export const highestScoreFrom = (words) => {
-  // Implement this method for wave 4
+	const wordScores = [];
+	const result = {};
+
+	for (let word of words) {
+		const score = scoreWord(word);
+		wordScores.push(score);
+	}
+	const highestScore = Math.max.apply(Math, wordScores);
+
+	for (let word of words) {
+		if (word.length === 10 && scoreWord(word) === highestScore) {
+			const winningWord = word;
+			result["word"] = winningWord;
+			result["score"] = highestScore;
+			{
+				break;
+			}
+		} else {
+			const sortedWordList = words.sort(function (a, b) {
+				return a.length - b.length;
+			});
+			for (let word of sortedWordList) {
+				if (scoreWord(word) === highestScore) {
+					const winningWord = word;
+					result["word"] = winningWord;
+					result["score"] = highestScore;
+					{
+						break;
+					}
+				}
+			}
+		}
+	}
+	return result;
```

Comment:
I understand what you're trying to do here, but creating a variety of extra objects and lists tends to overcomplicate the function just a bit. A more efficient approach would be to start with an object that holds the first word and its score. Then you can go ahead and loop through each of the words and judge it against what is in the object. If you do that in place, it really reduces the complexity! This approach might look like:

      export const highestScoreFrom = (words) => {
        let winningWord = {
          word: null,
          score: null
        }

        for (let word of words){

          let thisWord = {
            word: word,
            score: scoreWord(word)
          }

          if (thisWord.score > winningWord.score){
            winningWord = thisWord
          } else if (thisWord.score === winningWord.score){
            const isBestTen = winningWord.word.length === 10;
            const isWordTen = thisWord.word.length === 10;
            const isWordShorter = thisWord.word.length < winningWord.word.length;

            if ((isWordTen && !isBestTen) || (isWordShorter && !isBestTen)){
              winningWord = thisWord;
            }
          }
        }
        return winningWord;
      }